lib/spack/spack/cmd/clean.py

```
@@ -88,6 +88,26 @@ def remove_python_cache():
                     shutil.rmtree(dname)
 
 
+def remove_downloads():
+    """Remove all (relevant) cached download files.
+
+    This removes files in the fetch cache and, if relevant, the active
+    environment's included configuration files cache."""
+    tty.msg("Removing cached downloads")
+    spack.caches.fetch_cache.destroy()
+    what = "cached, included config files"
+    env = spack.environment.active_environment()
+    if env:
+        tty.msg("Removing {0} for '{1}' environment".format(what, env.name))
+        spack.environment.clear_included_configs()
```

Usually the clean options are global (at least without further specification) e.g. "remove *all* downloaded tarballs" if you run `spack clean -d`. For directory-based environments, I think you're doing the best that can be done (since they are not tracked by Spack). However, for Spack-managed environments I think it could make sense to have this option clear cached include files for all of them.

If - for simplicity of implementation - this should work on just the current active environment, then IMO the `else:` part of this statement should raise an error (there is a `require_active_environment` or similarly-named function available for this purpose).